mempool/txpool/legacypool/legacypool.go

```diff
@@ -1590,22 +1628,51 @@ func (pool *LegacyPool) demoteUnexecutables() {
 			log.Trace("Removed old pending transaction", "hash", hash)
 		}
 		// Drop all transactions that are too costly (low balance or out of gas), and queue any invalids back for later
-		drops, invalids := list.Filter(pool.currentState.GetBalance(addr), gasLimit)
+		drops, costInvalids := list.CostFilter(pool.currentState.GetBalance(addr), gasLimit)
```

Comment:
Ditto, could we just remove this and use the recheck?

Would also be worth noting that it seems we do a lot of duplicate checks, even in the antehandler, so we should track removing those in a separate issue.